packages/danger/src/PullRequestManager.ts

```diff
@@ -0,0 +1,174 @@
+    };
+
+    parseLine(this.pullRequest.title);
+
+    const changelogSection = this.pullRequest.body.match(/#\schangelog(([^#]*?)\s?)*/i)?.[0];
+    if (changelogSection) {
+      changelogSection
+        .replace(/^-/, '')
+        .split('\n')
+        .slice(1)
+        .map(line => line.trim())
+        .filter(line => line.length > 0)
+        .forEach(parseLine);
+    }
+
+    return changelogEntries;
+  }
+
+  async createOrUpdatePRAsync(
+    missingEntries: { packageName: string; content: string }[]
+  ): Promise<PullRequest | null> {
+    const dangerHeadRef = `@danger/add-missing-changelog-to-${this.pullRequest.number}`;
+    const dangerBaseRef = this.pullRequest.head.ref;
+
+    const fileMap = missingEntries.reduce(
+      (prev, current) => ({
+        ...prev,
+        [getPackageChangelogRelativePath(current.packageName)]: current.content,
+      }),
+      {}
+    );
+
+    await this.githubApi.createOrUpdateBranchFromFileMap(fileMap, {
+      baseBranchName: dangerBaseRef,
+      branchName: dangerHeadRef,
+      message: `${dangerTags} ${dangerMessage}`,
+    });
+
+    const prs = await this.githubApi.getOpenPRs({
+      fromBranch: dangerHeadRef,
+      toBranch: dangerBaseRef,
+    });
+
+    if (prs.length > 1) {
+      warn("Couldn't find the correct pull request. Too many open ones.");
+      return null;
+    }
+
+    if (prs.length === 1) {
+      return prs[0];
+    }
+
+    return this.githubApi.openPR({
+      fromBranch: dangerHeadRef,
+      toBranch: dangerBaseRef,
+      title: `${dangerTags} ${dangerMessage} to #${this.pullRequest.number}`,
+      body: `${dangerMessage} to #${this.pullRequest.number}`,
+    });
+  }
+
+  private parseTagsFromLine(
+    line: string
+  ): { packageName: string | typeof DEFAULT_CHANGELOG_ENTRY_KEY; type: ChangelogEntryType } | null {
+    const result: {
+      packageName: string | typeof DEFAULT_CHANGELOG_ENTRY_KEY;
+      type: ChangelogEntryType;
+    } = {
+      type: DEFAULT_ENTRY_TYPE,
+      packageName: DEFAULT_CHANGELOG_ENTRY_KEY,
+    };
+
+    const tags = line.match(/\[[^\]]*\]/g);
+    if (!tags) {
+      return result;
+    }
+
+    for (const tag of tags) {
+      const entryType = parseEntryType(tag);
+      if (entryType) {
+        result.type = Math.max(result.type, entryType);
+      } else if (isExpoPackage(tag)) {
+        result.packageName = tag.replace(/\[|\]/g, '').trim();
+      }
+    }
+
+    return result;
+  }
+}
+
+export function createPullRequestManager(api: Octokit, pr: PullRequest): PullRequestManager {
+  return new PullRequestManager(
+    pr,
+    new GithubApiWrapper(api, pr.base.user.login, pr.base.repo.name)
+  );
+}
+
+function parseEntryType(tag: string): ChangelogEntryType | null {
+  switch (true) {
+    case /\b(bug|fix|bugfix|bug-fix)\b/i.test(tag):
+      return ChangelogEntryType.BUG_FIXES;
+    case /\b(feat|features?)\b/i.test(tag):
+      return ChangelogEntryType.NEW_FEATURES;
+    case /\b(break(ing)?)\b/i.test(tag):
```

Comment:
Sorry for nitpicking, but those cases still needs to in reversed order. There are some rare cases that might happen: `[breaking feature]`, `[breaking fix]` 🙂